sql/policy_internal.sql

```diff
@@ -76,15 +78,35 @@ BEGIN
       AND (ch.status = 0 OR ch.status = 3)
   LOOP
     IF chunk_rec.status = 0 THEN
-       PERFORM @extschema@.compress_chunk( chunk_rec.oid );
+      BEGIN
+        PERFORM @extschema@.compress_chunk( chunk_rec.oid );
+        EXCEPTION WHEN OTHERS THEN
+        GET STACKED DIAGNOSTICS
+            _message = MESSAGE_TEXT,
+            _detail = PG_EXCEPTION_DETAIL;
+        RAISE WARNING 'compressing chunk "%" failed when compression policy is executed', chunk_rec.oid::regclass::text
+            USING DETAIL = format('Message: (%s), Detail: (%s).', _message, _detail),
+                  ERRCODE = sqlstate;
+      END;
     ELSIF chunk_rec.status = 3 AND recompress_enabled IS TRUE THEN
-       PERFORM @extschema@.decompress_chunk(chunk_rec.oid, if_compressed => true);
-       COMMIT;
-       -- SET LOCAL is only active until end of transaction.
-       -- While we could use SET at the start of the function we do not
-       -- want to bleed out search_path to caller, so we do SET LOCAL
-       -- again after COMMIT
-       PERFORM @extschema@.compress_chunk(chunk_rec.oid);
+      BEGIN
+        PERFORM @extschema@.decompress_chunk(chunk_rec.oid, if_compressed => true);
+        EXCEPTION WHEN OTHERS THEN
```

Comment:
```suggestion
      EXCEPTION WHEN OTHERS THEN
```
Nit. Just an indentation issue. It's up to you.